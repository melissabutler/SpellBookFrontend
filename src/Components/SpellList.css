.SpellList {
    display: inline;
    margin: auto;
    width: 55vw;
    list-style: none;
    align-items: center;
}

.SpellList-Link {
    color: black;
}