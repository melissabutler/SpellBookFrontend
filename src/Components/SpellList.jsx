import { useContext } from "react";
import { Routes, Route, Navigate } from 'react-router-dom';

import CurrentUserContext from "./currentUserContext";

//Components

import Home from "./Components/Home";
import LoginForm from "./Components/FormLogin";
import SignupForm from "./Components/FormSignup";
import LogOut from "./Components/FormLogOut";
import Profile from "./Components/Profile"

import SpellList from "./Components/SpellList";
import SpellCard from "./Components/SpellCard";

import CharacterList from "./Components/CharacterList";
import CharacterProfile from "./Components/CharacterProfile";

const RoutesList = ({login, 
                        signUp, 
                        logOut,
                        getUser,
                        editUser, 
                        getCharacter, 
                        createCharacter, 
                        editCharacter, 
                        deleteCharacter,
                        assignSpell,
                        unassignSpell,
                        getClass
}) => {
    const currentUser = useContext(CurrentUserContext);

    return (
        <Routes>
            {/* USER ROUTES */}
            <Route
                path='/'
                element={<Home />}
                />
            <Route
                path='/login'
                element={<LoginForm login={login}/>}
                />
            <Route 
                path='/signup'
                element={<SignupForm signUp={signUp} />}
                />
            <Route 
                path='/logout'
                element={currentUser ? (<LogOut logOut={logOut}/>) : <Navigate replace to="/"/>}
                />

            {/* Spell Routes */}
            <Route 
                path="/spells"
                element={<SpellList />}
                />
            <Route 
                path="/spells/:idx"
                element={<SpellCard assignSpell={assignSpell}
                                    getUser={getUser}
                                    unassignSpell={unassignSpell}
                                    details={true}
                                    charProfile={false}/>}
            />
            <Route 
                path="/characters"
                element={currentUser ? (<CharacterList createCharacter={createCharacter} getUser={getUser}/>) : <Navigate replace to="/"/>}
                /> 


            <Route 
                path="/characters/:id"
                element={ currentUser ? (<CharacterProfile getCharacter={getCharacter} 
                                                            unassignSpell={unassignSpell}
                                                            editCharacter={editCharacter} 
                                                            deleteCharacter={deleteCharacter}
                                                            getClass={getClass}
                    />) : <Navigate replace to="/"/>}
                /> 


            <Route 
                path='/profile'
                element={currentUser ? (<Profile editUser={editUser}/>) : <Navigate replace to="/"/>}
                />

            
            
        </Routes>
    )
}

export default RoutesList;