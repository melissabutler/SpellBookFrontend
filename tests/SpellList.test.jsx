import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import SpellList from "../src/Components/SpellList";
import SpellBookApi from "../api";



describe("SpellList", function() {

    it('renders SpellList', () => {
        render(
            <SpellList />
        )
        screen.debug();
    });

    it("matches snapshot", function() {
        const { asFragment } = render(<SpellList />);
        expect(asFragment()).toMatchSnapshot();
    })
        
    
})
