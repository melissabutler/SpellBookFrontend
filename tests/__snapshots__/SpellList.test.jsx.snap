// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpellList > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SpellList container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="col"
      >
        <h1
          class="SpellList-title"
        >
          Spell List
        </h1>
      </div>
      <div
        class="col"
      />
    </div>
    <div
      class="SpellList-search row"
    >
      <div
        class="col"
      />
      <div
        class="col-7"
      >
        <div
          class="ListSearch container"
        >
          <form>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Select Class
                </button>
                <select>
                  <option
                    value=""
                  >
                    All
                  </option>
                  <option
                    value="bard"
                  >
                    Bard
                  </option>
                  <option
                    value="cleric"
                  >
                    Cleric
                  </option>
                  <option
                    value="druid"
                  >
                    Druid
                  </option>
                  <option
                    value="paladin"
                  >
                    Paladin
                  </option>
                  <option
                    value="ranger"
                  >
                    Ranger
                  </option>
                  <option
                    value="sorceror"
                  >
                    Sorceror
                  </option>
                  <option
                    value="warlock"
                  >
                    Warlock
                  </option>
                  <option
                    value="wizard"
                  >
                    Wizard
                  </option>
                </select>
              </div>
              <div
                class="col"
              >
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Select Level
                </button>
                <select>
                  <option
                    value=""
                  >
                    All
                  </option>
                  <option
                    value="0"
                  >
                    0
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                </select>
              </div>
            </div>
            <button
              class="btn btn-primary"
              type="button"
            >
              Search
            </button>
            <div
              class="row"
            >
              <div
                class="col"
              />
              <div
                class="col-7"
              >
                <input
                  name="term"
                  placeholder="Search for a specific spell"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col"
              />
            </div>
          </form>
        </div>
      </div>
      <div
        class="col"
      />
    </div>
    <div
      class="SpellList-list row"
    >
      <div
        class="col"
      />
      <div
        class="col"
      />
      <div
        class="col"
      />
    </div>
  </div>
</DocumentFragment>
`;
